spec/prog/postgres/converge_postgres_resource_spec.rb

```diff
@@ -3,403 +3,484 @@
     end
 
     it "starts upgrade when not started" do
-      expect(candidate.vm.sshable).to receive(:d_check).with("upgrade_postgres").and_return("NotStarted")
-      expect(postgres_resource).to receive(:target_version).and_return("17")
-      expect(candidate.vm.sshable).to receive(:d_run).with("upgrade_postgres", "sudo", "postgres/bin/upgrade", "17")
+      expect(sshable).to receive(:d_check).with("upgrade_postgres").and_return("NotStarted")
+      expect(sshable).to receive(:d_run).with("upgrade_postgres", "sudo", "postgres/bin/upgrade", "17")
       expect { nx.upgrade_standby }.to nap(5)
     end
 
     it "naps if status of the upgrade is unknown" do
-      expect(candidate.vm.sshable).to receive(:d_check).with("upgrade_postgres").and_return("Unknown")
+      expect(sshable).to receive(:d_check).with("upgrade_postgres").and_return("Unknown")
       expect { nx.upgrade_standby }.to nap(5)
     end
   end
 
   describe "#update_metadata" do
-    let(:candidate) { instance_double(PostgresServer) }
-    let(:new_timeline) { instance_double(Strand, id: "new_timeline_id") }
+    let(:candidate) { create_server(version: "16") }
 
     before do
+      strand.update(label: "update_metadata")
       allow(nx).to receive(:upgrade_candidate).and_return(candidate)
     end
 
     it "creates new timeline and updates candidate server metadata and hops to recycle_representative_server" do
-      expect(Prog::Postgres::PostgresTimelineNexus).to receive(:assemble).with(location_id: anything).and_return(new_timeline)
-      expect(postgres_resource).to receive(:location_id).and_return("location_id")
-      expect(postgres_resource).to receive(:target_version).and_return("17")
-
-      expect(candidate).to receive(:update).with(version: "17", timeline_id: "new_timeline_id", timeline_access: "push")
-      expect(candidate).to receive(:incr_refresh_walg_credentials)
-      expect(candidate).to receive(:incr_configure)
-      expect(candidate).to receive(:incr_restart)
+      expect(Prog::Postgres::PostgresTimelineNexus).to receive(:assemble).with(location_id: location_id).and_call_original
 
       expect { nx.update_metadata }.to hop("wait_upgrade_candidate")
+
+      candidate.reload
+      expect(candidate.version).to eq("17")
+      expect(candidate.timeline_access).to eq("push")
+      expect(candidate.refresh_walg_credentials_set?).to be true
+      expect(candidate.configure_set?).to be true
+      expect(candidate.restart_set?).to be true
     end
   end
 
   describe "#wait_upgrade_candidate" do
+    before do
+      strand.update(label: "wait_upgrade_candidate")
+    end
+
     it "hops to recycle_representative_server when candidate is ready" do
-      candidate = instance_double(PostgresServer, restart_set?: false, strand: instance_double(Strand, label: "wait"))
+      candidate = create_server(version: "16")
+      candidate.strand.update(label: "wait")
       allow(nx).to receive(:upgrade_candidate).and_return(candidate)
```

Comment:
It looks like `upgrade_candidate` could be democked. Similar issue in other specs.